toolkit/tools/bldtracker/bldtracker.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Tool to initialize a csv file or record a new timestamp
+// for shell scripts during the image-building process.
+
+package main
+
+import (
+	"os"
+	"time"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/timestamp"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app          = kingpin.New("bldtracker", "A tool that helps track build time of different steps in the makefile.")
+	scriptName   = app.Flag("script-name", "The name of the current tool.").Required().String()
+	stepName     = app.Flag("step-name", "The name of the current step.").Required().String()
+	actionName   = app.Flag("action-name", "The name of the current action.").Default("").String()
+	dirPath      = app.Flag("dir-path", "The folder that stores timestamp csvs.").Required().ExistingDir() // currently must be absolute
+	logPath      = app.Flag("log-path", "Directory for log files").Required().ExistingDir()
+	validModes   = []string{"n", "r"}
+	mode         = app.Flag("mode", "The mode of this tool. Could be 'initialize' ('n') or 'record'('r').").Required().Enum(validModes...)
+	completePath string
+)
+
+func main() {
+	app.Version(exe.ToolkitVersion)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	logger.InitBestEffort(*logPath+"/chroot_timestamp.log", "trace")
+
+	// Construct the csv path.
+	completePath = *dirPath + "/" + *scriptName + ".csv"
```

Comment:
In general it's advised to use built-in libraries or constants to add file path separators. This makes the code easier to port, if needed, because the build-in function/constant for the other OS will always correctly build the path for us. In this case:

```suggestion
	completePath = filepath.Join(*dirPath, *scriptName + ".csv")
```

Please also update the path constructions elsewhere, if applicable.